backend/src/main/java/com/woowacourse/levellog/team/application/TeamService.java

```diff
@@ -57,123 +42,34 @@ public TeamStatusDto findStatus(final Long teamId) {
     }
 
     public InterviewRoleDto findMyRole(final Long teamId, final Long targetMemberId, final Long memberId) {
-        final Team team = teamRepository.getTeam(teamId);
-        final Participants participants = new Participants(participantRepository.findByTeam(team));
-        final InterviewRole interviewRole = participants.toInterviewRole(teamId, targetMemberId, memberId,
-                team.getInterviewerNumber());
+        final Team team = teamRepository.getTeamWithParticipants(teamId);
+        final InterviewRole interviewRole = team.getInterviewRole(targetMemberId, memberId);
 
         return InterviewRoleDto.from(interviewRole);
     }
 
     @Transactional
     public void update(final TeamWriteDto request, final Long teamId, final Long memberId) {
-        final Team team = teamRepository.getTeam(teamId);
-        validateHostAuthorization(memberId, team);
-        team.update(request.toEntity(team.getProfileUrl()), timeStandard.now());
-
-        final Participants participants = createParticipants(team, memberId, request.getParticipantIds(),
-                request.getWatcherIds());
-        team.validParticipantNumber(participants.size());
-        participantRepository.deleteByTeam(team);
-        participantRepository.saveAll(participants.getValues());
+        final Team team = teamRepository.getTeamWithParticipants(teamId);
+        team.validateHostAuthorization(memberId);
+
+        team.update(request.toTeamDetail(team.getProfileUrl()), request.toParticipantsIngredient(memberId),
+                timeStandard.now());
```

Comment:
```suggestion
        teamRepository.getTeamWithParticipants(teamId)
                .update(request.toTeamDetail(), request.toParticipantsIngredient(memberId), timeStandard.now());
```

`validateHostAuthorization`와 같은 검증 메서드를 도메인 객체 내부로 감추는건 어딸까요?
해당 검증 메서드는 도메인 객체의 상태를 변경하기 전에 호출하는 메서드이므로 해당 메서드를 application 계층에 노출할 필요는 없어보여요.

+) **`close`, `delete`역시 마찬가지네요** :)

Reply:
동의합니다. 고쳐볼게요. 😀

Reply:
고민점
- validateHostAuthorization를 위한 memberId를 ParticipantsIngredient에서 getter로 얻어와야 함
- AdminService.closeTeam 때문에 팀의 hostId를 구해오는 로직이 추가되어야 함